src/pkg/packager/publish.go

```diff
@@ -84,26 +85,42 @@ func (p *Packager) Publish() (err error) {
 	}
 
 	if p.cfg.CreateOpts.IsSkeleton {
-		cwd, err := os.Getwd()
-		if err != nil {
-			return err
+		if err := os.Chdir(p.cfg.CreateOpts.BaseDir); err != nil {
+			return fmt.Errorf("unable to access directory %q: %w", p.cfg.CreateOpts.BaseDir, err)
 		}
-		if err := p.cdToBaseDir(p.cfg.CreateOpts.BaseDir, cwd); err != nil {
+
+		c := creator.New(p.cfg.CreateOpts, "")
+
+		loadedPkg, warnings, err := c.LoadPackageDefinition(p.layout)
+		if err != nil {
 			return err
 		}
-		if err := p.load(); err != nil {
+
+		p.warnings = append(p.warnings, warnings...)
+
+		if err := c.Assemble(p.layout, loadedPkg); err != nil {
 			return err
 		}
-		if err := p.assembleSkeleton(); err != nil {
+
+		if err := c.Output(p.layout, loadedPkg); err != nil {
 			return err
 		}
+
+		p.cfg.Pkg = *loadedPkg
 	} else {
-		if err = p.source.LoadPackage(p.layout, false); err != nil {
+		if err := p.source.LoadPackage(p.layout, false); err != nil {
 			return fmt.Errorf("unable to load the package: %w", err)
 		}
-		if err = p.readZarfYAML(p.layout.ZarfYAML); err != nil {
+		if err := utils.ReadYaml(p.layout.ZarfYAML, &p.cfg.Pkg); err != nil {
 			return err
 		}
+
+		// Sign the package if a key has been provided
+		if p.cfg.PublishOpts.SigningKeyPath != "" {
+			if err := p.layout.SignPackage(p.cfg.PublishOpts.SigningKeyPath, p.cfg.PublishOpts.SigningKeyPassword); err != nil {
+				return err
+			}
```

Comment:
Why do we no longer sign skeleton packages?

Reply:
We do sign skeleton packages in `Output()` to remain consistent with normal packages being signed in `Output()`. It is being signed but the details aren't explicit when calling `Output()`